### PR TITLE
[Java][Client] Add Quarkus support

### LINKED ISSUE #8015: [REQ] Java MicroPofile Rest client with Jackson serialization?
### Is your feature request related to a problem? Please describe.

Quarkus, an up-and-coming Java Microprofile implementation will soon allow users to use JAX-RS with Jackson, instead of the standard JSONB as discussed [here](https://groups.google.com/g/quarkus-dev/c/RTnvH6LSdxs)

Currently this generator cannot generate Microprofile rest client code  based on JAX-RS+jackson code. I tried this (in my Gradle):
```
   ...
    openApiGenerate {
        generatorName = "java"       
        configOptions = [
                dateLibrary: "java8",
                java8: "true",
                library:"microprofile",
                serializationLibrary:"jackson"
        ]
     
    }
```
but the generated code is still JAX-RS/JSONB, not JAX-RS/Jackson

## Describe the solution you'd like

This should do the trick:
```
 openApiGenerate {
        generatorName = "java"       
        configOptions = [
                ...
                library:"microprofile",
                serializationLibrary:"jackson"
        ]
```

### PR DESCRIPTION
**Pull request by djnalluri**
This PR adds support for Quarkus in the Microprofile library for the Java client. These changes, for the most part, should not affect any existing usage with the exception of `disableMultipart` being renamed to `enableMultipart` and new annotations being used when `withXml` is enabled. To generate for Quarkus, set the library to `microprofile` and add the additional property named `microprofileFramework` set to `quarkus`.

Summary of changes:
- Added JSON-B as supported serialization library
- Added CLI option for controlling output of Swagger annotations
- Merged Microprofile-specific model templates into the general Java client templates
- Removed unused Microprofile templates
- Fixes #8015
- Support for using 'asyncNative' property to toggle usage of CompletionStage or Mutiny (void return not supported)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Technical Committee:
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @nmuesch (2021/01)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 1]
```diff
@@ -566,11 +602,19 @@ public void processOpts() {
         } else if (SERIALIZATION_LIBRARY_GSON.equals(getSerializationLibrary())) {
             additionalProperties.put(SERIALIZATION_LIBRARY_GSON, "true");
             additionalProperties.remove(SERIALIZATION_LIBRARY_JACKSON);
+            additionalProperties.remove(SERIALIZATION_LIBRARY_JSONB);
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:605]: this is the place where you should use switch/case for these instead adding another `else if`

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 2]
```diff
@@ -746,18 +790,38 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
                 model.imports.add("JsonWriter");
                 model.imports.add("IOException");
             }
+            if (additionalProperties.containsKey(SERIALIZATION_LIBRARY_JSONB)) {
+                //TODO: Fill this out
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:794]: todo

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 3]
```diff
@@ -746,18 +790,38 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
...
+            }
         }
-        if (MICROPROFILE.equals(getLibrary())) {
-            model.imports.remove("ApiModelProperty");
-            model.imports.remove("ApiModel");
-            model.imports.remove("JsonSerialize");
-            model.imports.remove("ToStringSerializer");
+//        if (MICROPROFILE.equals(getLibrary())) {
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:806]: revert commented out or fix and remove

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 4]
```diff
@@ -974,6 +1042,8 @@ public void setSerializationLibrary(String serializationLibrary) {
             this.serializationLibrary = SERIALIZATION_LIBRARY_JACKSON;
         } else if (SERIALIZATION_LIBRARY_GSON.equalsIgnoreCase(serializationLibrary)) {
             this.serializationLibrary = SERIALIZATION_LIBRARY_GSON;
+        } else if (SERIALIZATION_LIBRARY_JSONB.equalsIgnoreCase(serializationLibrary)) {
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:1045]: another place for switch/case

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 5]
```diff
@@ -1011,4 +1081,9 @@ public void addImportsToOneOfInterface(List<Map<String, String>> imports) {
             }
         }
     }
+
+    @Override
+    public Map<String, String> typeMapping() {
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:1086]: I dont think this override is needed here. `typeMapping()` is defined as public method in `DefaultCodeGen` and this doesnt change any behaviour. it returns the same `typeMapping` hashmap.

File: samples/client/petstore/java/microprofile-rest-client/src/main/java/org/openapitools/client/model/User.java [hunk 6]
```diff
@@ -44,188 +45,229 @@
...
-    this.username = username;
+
+  public User firstName(String firstName) {
+    
+    this.firstName = firstName;
     return this;
   }
 
- /**
+   /**
```
Comment (agilob) [inline samples/client/petstore/java/microprofile-rest-client/src/main/java/org/openapitools/client/model/User.java:105]: extra whitespace adds chaos to review

File: samples/client/petstore/java/microprofile-rest-client/src/main/java/org/openapitools/client/model/User.java [hunk 7]
```diff
@@ -44,188 +45,229 @@
...
    * Get firstName
    * @return firstName
   **/
+  @javax.annotation.Nullable
```
Comment (agilob) [inline samples/client/petstore/java/microprofile-rest-client/src/main/java/org/openapitools/client/model/User.java:109]: this should land in its own PR

File: samples/client/petstore/java/microprofile-rest-client/src/main/java/org/openapitools/client/model/Pet.java [hunk 8]
```diff
@@ -250,11 +298,12 @@ public String toString() {
    * Convert the given object to string with each line indented by 4 spaces
    * (except the first line).
    */
-  private static String toIndentedString(Object o) {
+  private String toIndentedString(Object o) {
```
Comment (agilob) [inline samples/client/petstore/java/microprofile-rest-client/src/main/java/org/openapitools/client/model/Pet.java:301]: this is potentially a breaking change with no fallback config

File: samples/client/petstore/java/microprofile-rest-client/src/main/java/org/openapitools/client/model/Tag.java [hunk 9]
```diff
@@ -25,66 +27,85 @@
...
   public void setName(String name) {
     this.name = name;
   }
 
-  public Tag name(String name) {
-    this.name = name;
-    return this;
+
+  @Override
+  public boolean equals(Object o) {
```
Comment (agilob) [inline samples/client/petstore/java/microprofile-rest-client/src/main/java/org/openapitools/client/model/Tag.java:88]: This also belongs in its own PR, only adds chaos to this review that's long already

**Comment by agilob** [comment]
As it's a good effort I think this PR has way too many changes to go all in, it should be at last 4-7. There are at least 2 breaking changes in here as well, rename of config flag, change of static method (which looks to be static intentionally).

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java (hunk 2 above)
Comment (djnalluri) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:794]: I left it intentionally blank as a marker for future optimization. But indeed, it should be removed or stated to be so.

**Comment by djnalluri** [comment]
I'm closing this PR and will break it up into multiple PR's as suggested. Thanks for the feedback.

**Comment by melloware** [comment]
@djnalluri was this ever implemented? I am looking for Quarkus support of Microprofile.

**Comment by djnalluri** [comment]
This hasn't been implemented yet. The series of PR's I was hoping to break things into are layered on each other. Progress on this is stuck on #8801. You could check out the Quarkiverse plugin. It uses OpenAPI generator under the hood. https://github.com/quarkiverse/quarkus-openapi-generator

**Comment by melloware** [comment]
OK you rock this Quarkiverse plugin looks like exactly what I am looking for.
